google/cloud/aiplatform/experimental/vertex_model/serializers/pandas.py

```diff
@@ -74,4 +79,29 @@ def _serialize_dataframe(artifact_uri: str, obj: pd.DataFrame, temp_dir: str) ->
 def _deserialize_dataframe(artifact_uri: str) -> str:
     """Provides out-of-the-box deserialization after training and prediction is complete"""
 
-    raise NotImplementedError
+    gcs_bucket, gcs_blob = utils.extract_bucket_and_prefix_from_gcs_path(artifact_uri)
+
+    client = storage.Client(
+        project=initializer.global_config.project,
+        credentials=initializer.global_config.credentials,
+    )
+
+    bucket = client.bucket(gcs_bucket)
+    blob = bucket.blob(gcs_blob)
+    df = pd.DataFrame()
```

Comment:
It doesn't seem like this is needed.

Reply:
The blob object is used to download the file from GCS.